### PR TITLE
[docs] update automatic code formatting info to clang-format

### LINKED ISSUE #3834: [docs] Update Style wrt Code Formatting
The following section of the style guide needs to be updated

https://github.com/PointCloudLibrary/pcl/blame/master/doc/advanced/content/pcl_style_guide.rst#L317-L377

Currently the code style we're enforcing is with clang-format, whose config file is also picked up by most IDEs. The remaining style files which were previously distributed should be considered deprecated and not distributed anymore.

Currently, to format code, most IDEs parse clang-format files (ours is [here](https://github.com/PointCloudLibrary/pcl/blob/master/.clang-format)). On Linux and Mac, there's always the option to run
```
$ make format
```
If clang-format is installed and detected during configuration.

### PR DESCRIPTION
**Pull request by divmadan**
closes #3834

### CODE REVIEW

**Comment by divmadan** [comment]
Suggestions/improvements?

File: doc/advanced/content/pcl_style_guide.rst [hunk 1]
```diff
@@ -317,64 +317,35 @@ indentation level of the class body and members affected by these qualifiers are
...
 
-2.6.1. Emacs
-""""""""""""
+The style rules mentioned in this document are enforced via `PCL's .clang-format file
+<https://github.com/PointCloudLibrary/pcl/blob/master/.clang-format>`_.
 
-You can use the following `PCL C/C++ style file
-<https://raw.githubusercontent.com/PointCloudLibrary/pcl/master/doc/advanced/content/files/pcl-c-style.el>`_,
-download it to some known location and then:
+For the integration of clang-format with various text editors and IDE's, refer `this
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:325]:
```suggestion
For the integration of clang-format with various text editors and IDE's, refer to this `page
```

File: doc/advanced/content/pcl_style_guide.rst [hunk 2]
```diff
@@ -317,64 +317,35 @@ indentation level of the class body and members affected by these qualifiers are
...
-"""""""""""""""""
+Sample command line usage from the PCL source directory:
 
-You can find a semi-finished config for `Uncrustify <http://uncrustify.sourceforge.net/>`_ `here
-<http://dev.pointclouds.org/attachments/download/537/uncrustify.cfg>`_
+.. code-block:: shell
 
-2.6.3 Eclipse
-"""""""""""""
+   $ sh ./.dev/format.sh `which clang-format` ./
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:342]: I would rather not expose this. Users should use the make target.

Comment (divmadan) [reply doc/advanced/content/pcl_style_guide.rst]: Done.

File: doc/advanced/content/pcl_style_guide.rst [hunk 3]
```diff
@@ -317,64 +317,28 @@ indentation level of the class body and members affected by these qualifiers are
...
+Details about the style options used can be found `here
+<https://clang.llvm.org/docs/ClangFormatStyleOptions.html>`_.
 
-::
+2.6.1. Script usage
+"""""""""""""""""""
 
-   (load-file "/location/to/pcl-c-style.el")
-   (add-hook 'c-mode-common-hook 'pcl-set-c-style)
+PCL also has a supporting format script which runs the clang-format on the whitelisted directories.
```
Comment (kunaltyagi) [inline doc/advanced/content/pcl_style_guide.rst:334]: It's not a script, rather a build target:

PCL also creates a build target 'format' to format the whitelisted directories using clang-format.

Comment (divmadan) [reply doc/advanced/content/pcl_style_guide.rst]: I was referring to `.dev/format.sh`, but your suggestion is much better :+1: 
Done.

**Pull request merged**